packages/@aws-cdk/core/lib/stack-synthesizers/default-synthesizer.ts

```diff
@@ -405,6 +405,26 @@ export class DefaultStackSynthesizer extends StackSynthesizer {
     return this._deployRoleArn;
   }
 
+  /**
+   * Returns the ARN of the image asset publishing Role.
+   */
+  public get imageAssetPublishingRoleArn(): string {
```

Comment:
There should be no need to make these public, as the information you need to determine the appropriate roles will be in the cloud assembly (and specifically, in the asset manifest of the Cloud Assembly for these two).

In fact, that is the correct flow of information. Reading from the synthesizer directly is a shortcut that may or may not be valid (and hence: dangerous).

Reply:
Ah yep. Sounds better.

Looks like for the assets action the `addToRolePolicy` calls would need to happen from in `addPublishCommand`, and in `addPublishCommand` the manifest would be read there to see what publish role is needed?

Not sure where the update pipeline action would get the cloud assembly from yet